ipa-core/src/protocol/dp/mod.rs

```diff
@@ -134,75 +214,61 @@ fn d_p(success_prob: f64) -> f64 {
 /// equation (7)
 #[allow(clippy::too_many_arguments)]
 #[allow(dead_code)]
-fn epsilon_constraint(
-    num_bernoulli: u32,
-    success_prob: f64,
-    delta: f64,
-    quantization_scale: f64,
-    dimensions: f64,
-    ell_1_sensitivity: f64,
-    ell_2_sensitivity: f64,
-    ell_infty_sensitivity: f64,
-) -> f64 {
+fn epsilon_constraint(num_bernoulli: u32, noise_params: &NoiseParams) -> f64 {
     let num_bernoulli_f64 = f64::from(num_bernoulli);
-    let first_term_num = ell_2_sensitivity * (2.0 * (1.25 / delta).ln()).sqrt();
-    let first_term_den =
-        quantization_scale * (num_bernoulli_f64 * success_prob * (1.0 - success_prob)).sqrt();
-    let second_term_num = ell_2_sensitivity * c_p(success_prob) * ((10.0 / delta).ln()).sqrt()
-        + ell_1_sensitivity * b_p(success_prob);
-    let second_term_den = quantization_scale
+    let first_term_num =
```

Comment:
I think it may be the time to start documenting what is going on here. This code is very dense and there are a lot of constants and math defined. where things like 1.0 with combination of probability can be explained, 10.0, 2.0, 3.0 are not so much I think.

You can improve this by defining constants that will help understand the code better.

```rust
   const MAX_PROBABILITY: f64 = 1.0; 
   const MAX_EPSILON: f64: 20.0
   ...
```

some of them seems to be re-usable in other parts of this module